packages/lib/src/accordion/AccordionItem.tsx

```diff
@@ -0,0 +1,245 @@
+import { ReactElement, useContext, useId, cloneElement, useMemo } from "react";
+import styled, { ThemeProvider } from "styled-components";
+import HalstackContext from "../HalstackContext";
+import { AccordionItemProps } from "./types";
+import DxcIcon from "../icon/Icon";
+import DxcFlex from "../flex/Flex";
+import DxcContainer from "../container/Container";
+import React from "react";
+import AccordionContext from "./AccordionContext";
+
+const AccordionItem = ({
+  label = "",
+  subLabel = "",
+  badge,
+  statusLight,
+  icon,
+  assistiveText = "",
+  disabled = false,
+  children,
+  tabIndex = 0,
+}: AccordionItemProps): JSX.Element => {
+  const id = useId();
+  const colorsTheme = useContext(HalstackContext);
+  const { activeIndex, handlerActiveChange, index, independent } = useContext(AccordionContext) ?? {};
+  const isItemExpanded = useMemo(() => {
+    return independent
+      ? activeIndex === index
+      : Array.isArray(activeIndex) && index !== undefined && activeIndex.includes(index);
+  }, [independent, activeIndex, index]);
+
+  const handleAccordionState = () => {
+    if (index !== undefined) handlerActiveChange?.(index);
+  };
+
+  return (
+    <ThemeProvider theme={colorsTheme.accordion}>
+      <AccordionContainer>
+        <AccordionTrigger
+          id={`accordion-${id}`}
+          onClick={disabled ? undefined : handleAccordionState}
+          disabled={disabled}
+          tabIndex={disabled ? -1 : tabIndex}
+          aria-expanded={isItemExpanded}
+          aria-controls={`accordion-panel-${id}`}
+        >
+          <DxcContainer width="100%" height="100%">
+            <DxcFlex gap="1.5rem">
+              <LeftSideContainer>
+                <DxcFlex gap="0.75rem">
+                  {icon ||
```

Comment:
As it stands, the conditional is not working properly (it renders the icon string instead of the svg). It's just a matter of how you are writing the conditional:

```typescript
{(icon || badge?.position === "before") && (
  <OptionalElement>
    {icon ? (
      <IconContainer disabled={disabled}>
        {typeof icon === "string" ? <DxcIcon icon={icon} /> : icon}
      </IconContainer>
    ) : (
      <StatusContainer subLabel={subLabel}>
        {disabled ? cloneElement(badge?.element as ReactElement, { color: "grey" }) : badge?.element}
      </StatusContainer>
    )}
  </OptionalElement>
)}
```

Reply:
Totally agree! The autosave put the `()` wrongly, thanks Iván!!